TO.py

```diff
@@ -125,33 +123,29 @@ def TO_Manipulator_Model(self,ICS, init_q0, init_q1, init_q2, init_v0, init_v1,
     
         ### Distence to target term (quadratic term + log valley centered at target) ###
-        m.dist_cost = sum((w_d*(((-7 + self.conf.l*(cos(m.q0[k]) + cos(m.q0[k]+m.q1[k]) + cos(m.q0[k]+m.q1[k]+m.q2[k])))-x_des)**2 + ((self.conf.l*(sin(m.q0[k]) + sin(m.q0[k]+m.q1[k]) + sin(m.q0[k]+m.q1[k]+m.q2[k])))-y_des)**2) - w_peak*(log(exp(alpha2*-(sqrt(((-7 + self.conf.l*(cos(m.q0[k]) + cos(m.q0[k]+m.q1[k]) + cos(m.q0[k]+m.q1[k]+m.q2[k])))-x_des)**2 +0.1) - 0.1 + sqrt(((self.conf.l*(sin(m.q0[k]) + sin(m.q0[k]+m.q1[k]) + sin(m.q0[k]+m.q1[k]+m.q2[k])))-y_des)**2 +0.1) - 0.1 -2*sqrt(0.1))) + 1)/alpha2)) for k in m.k) - (w_d*((-7 + self.conf.l*(cos(m.q0[0]) + cos(m.q0[0]+m.q1[0]) + cos(m.q0[0]+m.q1[0]+m.q2[0]))-x_des)**2 + (self.conf.l*(sin(m.q0[0]) + sin(m.q0[0]+m.q1[0]) + sin(m.q0[0]+m.q1[0]+m.q2[0]))-y_des)**2) - w_peak*(log(exp(alpha2*-(sqrt((-7 + self.conf.l*(cos(m.q0[0]) + cos(m.q0[0]+m.q1[0]) + cos(m.q0[0]+m.q1[0]+m.q2[0]))-x_des)**2 +0.1) - 0.1 + sqrt((self.conf.l*(sin(m.q0[0]) + sin(m.q0[0]+m.q1[0]) + sin(m.q0[0]+m.q1[0]+m.q2[0]))-y_des)**2 +0.1) - 0.1 -2*sqrt(0.1))) + 1)/alpha2))
+        m.dist_cost = sum((w_d*(((self.conf.x_base + self.conf.l*(cos(m.q0[k]) + cos(m.q0[k]+m.q1[k]) + cos(m.q0[k]+m.q1[k]+m.q2[k])))-x_des)**2 + ((self.conf.y_base + self.conf.l*(sin(m.q0[k]) + sin(m.q0[k]+m.q1[k]) + sin(m.q0[k]+m.q1[k]+m.q2[k])))-y_des)**2) - w_peak*(log(exp(alpha2*-(sqrt(((self.conf.x_base + self.conf.l*(cos(m.q0[k]) + cos(m.q0[k]+m.q1[k]) + cos(m.q0[k]+m.q1[k]+m.q2[k])))-x_des)**2 +0.1) - 0.1 + sqrt(((self.conf.y_base + self.conf.l*(sin(m.q0[k]) + sin(m.q0[k]+m.q1[k]) + sin(m.q0[k]+m.q1[k]+m.q2[k])))-y_des)**2 +0.1) - 0.1 -2*sqrt(0.1))) + 1)/alpha2)) for k in m.k) - (w_d*((self.conf.x_base + self.conf.l*(cos(m.q0[0]) + cos(m.q0[0]+m.q1[0]) + cos(m.q0[0]+m.q1[0]+m.q2[0]))-x_des)**2 + (self.conf.y_base + self.conf.l*(sin(m.q0[0]) + sin(m.q0[0]+m.q1[0]) + sin(m.q0[0]+m.q1[0]+m.q2[0]))-y_des)**2) - w_peak*(log(exp(alpha2*-(sqrt((self.conf.x_base + self.conf.l*(cos(m.q0[0]) + cos(m.q0[0]+m.q1[0]) + cos(m.q0[0]+m.q1[0]+m.q2[0]))-x_des)**2 +0.1) - 0.1 + sqrt((self.conf.y_base + self.conf.l*(sin(m.q0[0]) + sin(m.q0[0]+m.q1[0]) + sin(m.q0[0]+m.q1[0]+m.q2[0]))-y_des)**2 +0.1) - 0.1 -2*sqrt(0.1))) + 1)/alpha2))
 
-        # m.v = sum(((m.v0[k])**2 + (m.v1[k])**2 + (m.v2[k])**2)*(k/m.k[-1])**2 for k in m.k) 
+        ### Final velocity term ###
         m.v = (m.v0[N])**2 + (m.v1[N])**2 + (m.v2[N])**2
 
         m.obj = Objective(expr = (m.dist_cost + w_v*m.v + w_ob1*m.ell1_penalty + w_ob2*m.ell2_penalty + w_ob3*m.ell3_penalty + w_u*m.u_obj - 10000)/100, sense=minimize)
         return m
 
-    def TO_Manipulator_Solve(self, ep, state_prev, init_TO_controls, init_TO_states):
-        ''' Solve TO problem - manipulator '''                   
+    def TO_Manipulator_Solve(self, ep, init_rand_state, init_TO_states, init_TO_controls, NSTEPS_SH):
```

Comment:
This is just a TO solver, it doesn't know that the initial state is random, so I wouldn't call it `init_rand_state`